In [21]:
import pandas as pd

Download all 6.9 million Yelp Reviews

In [2]:
filename = "yelp_academic_dataset_review" + ".json"
path = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Datasets/" + filename

df_review = pd.read_json(path, lines=True, chunksize=100000)

chunks = []

for i, chunk in enumerate(df_review):
    chunks.append(chunk)
    # if i == 9:
    #     break
    
chunk_review = pd.concat(chunks, ignore_index=True)

chunk_review.shape

(6990280, 9)

Download all 150K Yelp Businesses

In [3]:
filename = "yelp_academic_dataset_business" + ".json"
path = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Datasets/" + filename

df_business = pd.read_json(path, lines=True, chunksize=100000)

business_chunks = []

for i, chunk in enumerate(df_business):
    business_chunks.append(chunk)
    
    # if i == 9:
    #     break
    
chunk_business = pd.concat(business_chunks, ignore_index=True)

chunk_business.shape

(150346, 14)

In [4]:
states = chunk_business['state'].value_counts()
print(states)

PA     34039
FL     26330
TN     12056
IN     11247
MO     10913
LA      9924
AZ      9912
NJ      8536
NV      7715
AB      5573
CA      5203
ID      4467
DE      2265
IL      2145
TX         4
CO         3
WA         2
HI         2
MA         2
NC         1
UT         1
MT         1
MI         1
SD         1
XMS        1
VI         1
VT         1
Name: state, dtype: int64


Filter by state of Florida

In [5]:
fl_business = chunk_business[chunk_business['state'] == 'FL']
fl_business.shape

(26330, 14)

In [6]:
categories = fl_business['categories'].value_counts()
print(categories[0:25])

Beauty & Spas, Nail Salons                            166
Nail Salons, Beauty & Spas                            140
Restaurants, Pizza                                    124
Restaurants, Chinese                                  113
Pizza, Restaurants                                    102
Chinese, Restaurants                                   94
Mexican, Restaurants                                   90
Beauty & Spas, Hair Salons                             83
Restaurants, Mexican                                   82
Food, Coffee & Tea                                     73
Hair Salons, Beauty & Spas                             70
Grocery, Food                                          67
Barbers, Beauty & Spas                                 65
Hotels, Hotels & Travel, Event Planning & Services     64
Automotive, Auto Repair                                63
Pets, Veterinarians                                    63
Auto Repair, Automotive                                62
Apartments, Ho

Filtering for Florida Restaurants Only

In [7]:
fl_restaurants = fl_business[fl_business['categories'].str.contains('restaurants', case=False, na=False)]
fl_restaurants.shape

(8731, 14)

Merging Florida Restaurants df with Reviews df

In [8]:
merged_df = pd.merge(fl_restaurants, chunk_review, on='business_id', how='inner', suffixes=('_business', '_reviews'))
merged_df.shape

(792133, 22)

In [20]:
filename = "FL_Restaurants_Reviews" + ".csv"
out_path = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Datasets/" + filename

merged_df.to_csv(out_path, index=False)

In [9]:
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 792133 entries, 0 to 792132
Data columns (total 22 columns):
 #   Column          Non-Null Count   Dtype         
---  ------          --------------   -----         
 0   business_id     792133 non-null  object        
 1   name            792133 non-null  object        
 2   address         792133 non-null  object        
 3   city            792133 non-null  object        
 4   state           792133 non-null  object        
 5   postal_code     792133 non-null  object        
 6   latitude        792133 non-null  float64       
 7   longitude       792133 non-null  float64       
 8   stars_business  792133 non-null  float64       
 9   review_count    792133 non-null  int64         
 10  is_open         792133 non-null  int64         
 11  attributes      791260 non-null  object        
 12  categories      792133 non-null  object        
 13  hours           770229 non-null  object        
 14  review_id       792133 non-null  obj

In [10]:
categories = merged_df['categories'].value_counts()
print(categories[0:25])

Mexican, Restaurants                                                                                                                                                                                              8384
Restaurants, Mexican                                                                                                                                                                                              6363
Restaurants, Pizza                                                                                                                                                                                                5403
Pizza, Restaurants                                                                                                                                                                                                5191
Italian, Restaurants                                                                                                                        

Top 25 Most Reviewed Restaurants in Tampa Bay Area

In [12]:
business_names = merged_df['name'].value_counts()
print(business_names[0:25])

Columbia Restaurant         4032
Datz                        3577
McDonald's                  3340
First Watch                 3230
Ulele                       3179
Bern's Steak House          3028
Taco Bus                    2679
Noble Crust                 2360
Frenchy's Rockaway Grill    2335
Capital Tacos               2323
Ford's Garage               2305
Chick-fil-A                 2148
Chipotle Mexican Grill      2035
Bodega                      2029
Chili's                     1965
Oxford Exchange             1938
Lolis Mexican Cravings      1889
Tijuana Flats               1882
Bonefish Grill              1875
Clear Sky Cafe              1869
Acropolis Greek Taverna     1847
Outback Steakhouse          1819
Panera Bread                1756
Buffalo Wild Wings          1637
Bahama Breeze               1637
Name: name, dtype: int64


Download All 200K Photo Information Dataset

In [13]:
filename = "photos" + ".csv"
path = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Datasets/" + filename

df_photos = pd.read_csv(path, chunksize=100000)

photos_chunks = []

for i, chunk in enumerate(df_photos):
    photos_chunks.append(chunk)
    
    # if i == 9:
    #     break
    
chunk_photos = pd.concat(photos_chunks, ignore_index=True)

chunk_photos.shape

(200100, 4)

In [14]:
chunk_photos.head()

photo_id             business_id  \
0  zsvj7vloL4L5jhYyPIuVwg  Nk-SJhPlDBkAZvfsADtccA   
1  HCUdRJHHm_e0OCTlZetGLg  yVZtL5MmrpiivyCIrVkGgA   
2  vkr8T0scuJmGVvN2HJelEA  _ab50qdWOk0DdB6XOrBitw   
3  pve7D6NUrafHW3EAORubyw  SZU9c8V2GuREDN5KgyHFJw   
4  H52Er-uBg6rNrHcReWTD2w  Gzur0f0XMkrVxIwYJvOt2g   

                                             caption    label  
0  Nice rock artwork everywhere and craploads of ...   inside  
1                                                NaN  outside  
2                                     oyster shooter    drink  
3                                      Shrimp scampi     food  
4                                                NaN     food

In [15]:
labels = chunk_photos['label'].value_counts()
print(labels)

food       108152
inside      56031
outside     18569
drink       15670
menu         1678
Name: label, dtype: int64


Merge Florida Restaurant df with Photo Info df. Total of 30.4K rows

In [16]:
merged_photos_df = pd.merge(fl_restaurants, chunk_photos, on='business_id', how='inner')
merged_photos_df.shape

(30410, 17)

In [17]:
merged_photos_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 30410 entries, 0 to 30409
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   business_id   30410 non-null  object 
 1   name          30410 non-null  object 
 2   address       30410 non-null  object 
 3   city          30410 non-null  object 
 4   state         30410 non-null  object 
 5   postal_code   30410 non-null  object 
 6   latitude      30410 non-null  float64
 7   longitude     30410 non-null  float64
 8   stars         30410 non-null  float64
 9   review_count  30410 non-null  int64  
 10  is_open       30410 non-null  int64  
 11  attributes    30368 non-null  object 
 12  categories    30410 non-null  object 
 13  hours         29819 non-null  object 
 14  photo_id      30410 non-null  object 
 15  caption       15237 non-null  object 
 16  label         30410 non-null  object 
dtypes: float64(3), int64(2), object(12)
memory usage: 4.2+ MB


In [19]:
filename = "FL_Restaurants_Photos" + ".csv"
out_path = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Datasets/" + filename

merged_photos_df.to_csv(out_path, index=False)

Create set of Photo IDs and iterate through Extracted Images Directory which includes all 200K images and match with Photo ID list and then copy photos to Florida Photos Directory. The resulting output includes 30,410 images which matches the number of rows in the merged_photos_df and will be all the relevant Florida photos available.

In [18]:
import os
import shutil

photos_dir = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Photos/extracted_images"
output_dir = "C:/Users/tokud/OneDrive/Documents/Machine Learning - UCSD Extension Springboard Course/Capstone/Datasets/Florida Photos/"

if not os.path.exists(output_dir):
    os.makedirs(output_dir)

photo_ids = set(merged_photos_df['photo_id'].tolist())  


for image_file in os.listdir(photos_dir):
    if image_file.endswith('.jpg'):

        photo_id = os.path.splitext(image_file)[0]
        
        if photo_id in photo_ids:
            src_path = os.path.join(photos_dir, image_file)
            dest_path = os.path.join(output_dir, image_file)
            
            shutil.copy(src_path, dest_path)
            
